src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,42 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import duke.commands.Command;
+
+/**
+ * Represents the chatbot Duke.
+ */
+public class Duke {
+
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for Duke class.
+     *
+     * @throws IOException when an IO operation fails.
+     */
+    public Duke() throws IOException {
+        this.tasks = new TaskList();
+        this.ui = new Ui(new Scanner(System.in));
+    }
+
+    /**
+     * Method used to run the program.
```

Comment:
Consider adding `@param` and `@return` tags to JavaDoc comments for public methods?